src/controllers/auth.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable consistent-return */
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import model from '../models/index';
+
+exports.signup = async (req, res, next) => {
+  const {
+    username, email, confirmPassword,
+  } = req.body;
+
+  let { password } = req.body;
+
+  if (password !== confirmPassword) {
+    return res.status(401).json({ message: "passwords don't match" });
+  }
+
+  await model.User.findOne({
+    where: {
+      email,
+
+    },
+
+  }).then((user) => {
+    if (user) {
+      return res.status(403).json({
+        error: 'email is already taken!',
+      });
+    }
+  });
+
+  bcrypt.hash(password, 10, (err, hash) => {
+    password = hash;
+
+    model.User.create({
+      username,
+      email,
+      password,
+
+    }).then((user) => {
+      const { id } = user;
+      const token = jwt.sign({ id }, process.env.secret);
+      return res.status(200).json({
+        token,
+        user: username,
+      });
```

Comment:
1. Is `200` the correct code for successfully creating a resource?
2. The username should be encoded into the token (together with the id) and not in the response object. You should still return only the token.